core/modules/layout/layout.api.php

```diff
@@ -196,6 +196,191 @@ function hook_layout_renderer_info() {
+ * To respond to a block being added via Layout Builder, use hook_block_save().
+ *
+ * @param Layout $layout
+ *   The layout object to which the block is being added.
+ * @param Block $block
+ *   The block object which is being added.
+ *
+ * @see hook_block_save()
+ * @see hook_block_remove()
+ *
+ * @ingroup layout_api_hooks
+ */
+function hook_layout_block_add(Block $block, Layout $layout) {
+  if ($block->delta == 'my_block') {
+    config_set('my_module', 'custom_settings', $layout->content[$block->uuid]);
+  }
+}
+
+/**
+ * Respond to a layout being reverted.
+ *
+ * Layouts can be reverted only if the configuration is provided
+ * by a module. Layouts created in the Layout Builder User Interface
+ * cannot be reverted. 
+ * A layout revert operation results in deletion of the existing
+ * layout configuration and replacement with the default configuration
+ * from the providing module. 
+ * This hook is invoked from Layout::Revert() after a layout has been 
+ * reverted and the new configuration has been inserted into the live
+ * config directory and the layout cache has been cleared.
+ *
+ * @param Layout $layout
+ *   The old layout object that has just been deleted.
+ *
+ * @ingroup layout_api_hooks
+ */
+function hook_layout_revert(Layout $old_layout) {
+  if ($old_layout == 'my_layout') {
+    my_custom_function();
+  }
+  // Get the new (reverted) configuration.
+  $new_config = config_get('layout.layout.' . $old_layout->name);
+}
+
+/**
+ * Respond to a layout being deleted.
+ *
+ * This hook is invoked from Layout::Delete() after a layout has been 
+ * deleted.
+ *
+ * @param Layout $layout
+ *   The layout object that was deleted.
+ *
+ * @ingroup layout_api_hooks
+ */
+function hook_layout_delete(Layout $layout) {
+  if ($layout->path == 'my_path') {
+    my_custom_function();
+  }
+}
+
+/**
+ * Respond to a layout being enabled.
+ *
+ * This hook is invoked from Layout::Enable() after a layout has been 
+ * enabled.
+ *
+ * @param Layout $layout
+ *   The layout object that was enabled.
+ *
+ * @ingroup layout_api_hooks
+ */
+function hook_layout_enable(Layout $layout) {
+  if ($layout->path == 'my_path') {
+    my_custom_function();
+  }
+}
+
+/**
+ * Respond to a layout being disabled.
+ *
+ * This hook is invoked from Layout::Disable() after a layout has been 
+ * disabled.
+ *
+ * @param Layout $layout
+ *   The layout object that was disabled.
+ *
+ * @ingroup layout_api_hooks
+ */
+function hook_layout_disable(Layout $layout) {
+  if ($layout->path == 'my_path') {
+    my_custom_function();
+  }
+}
+
+/**
+ * Respond to updates to a layout.
+ *
+ * This hook is invoked from Layout::Save() after a layout has been saved
+ * to configuration.
+ *
+ * @param Layout $layout
+ *   The layout object that was saved.
+ *
+ * @ingroup layout_api_hooks
+ */
+function hook_layout_update(Layout $layout) {
+  if ($layout->path == 'my_path') {
+    my_custom_function();
+  }
+}
+
+/**
+ * Respond to insertion of a layout.
```

Comment:
Maybe instead of saying "insertion" say "initial creation" in this documentation? Layouts aren't really "inserted" in the traditional sense (into a database), but I could see the appeal in sticking with our current conventions for hook naming.